[ao][sparsity] adding several sparse quantized layers

HDCharles opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #81803
* #81802

Summary: This PR adds sparse+quantized versions of the following quantized layers

linear_relu
linear_relu (qat)
linear (qat)

also updated the default sparse+quantized mappings to utilize these new
layers and also updated the qat linear layer to maintain the
sparse_params which also solves the problem for the intrinsic qat layers.

Test Plan: python test/test_ao_sparsity.py TestQuantizedSparseKernels
python test/test_ao_sparsity.py TestQuantizedIntrinsicSparseKernels

Reviewers:

Subscribers:

Tasks:

Tags:

HDCharles committed:
[ao][sparsity] adding several sparse quantized layers

Summary: This PR adds sparse+quantized versions of the following quantized layers

linear_relu
linear_relu (qat)
linear (qat)

also updated the default sparse+quantized mappings to utilize these new
layers and also updated the qat linear layer to maintain the
sparse_params which also solves the problem for the intrinsic qat layers.

Test Plan: python test/test_ao_sparsity.py TestQuantizedSparseKernels
python test/test_ao_sparsity.py TestQuantizedIntrinsicSparseKernels

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity] adding several sparse quantized layers"


Summary: This PR adds sparse+quantized versions of the following quantized layers

linear_relu
linear_relu (qat)
linear (qat)

also updated the default sparse+quantized mappings to utilize these new
layers and also updated the qat linear layer to maintain the
sparse_params which also solves the problem for the intrinsic qat layers.

Test Plan: python test/test_ao_sparsity.py TestQuantizedSparseKernels
python test/test_ao_sparsity.py TestQuantizedIntrinsicSparseKernels

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity] adding several sparse quantized layers"


Summary: This PR adds sparse+quantized versions of the following quantized layers

linear_relu
linear_relu (qat)
linear (qat)

also updated the default sparse+quantized mappings to utilize these new
layers and also updated the qat linear layer to maintain the
sparse_params which also solves the problem for the intrinsic qat layers.

Test Plan: python test/test_ao_sparsity.py TestQuantizedSparseKernels
python test/test_ao_sparsity.py TestQuantizedIntrinsicSparseKernels

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity] adding several sparse quantized layers"


Summary: This PR adds sparse+quantized versions of the following quantized layers

linear_relu
linear_relu (qat)
linear (qat)

also updated the default sparse+quantized mappings to utilize these new
layers and also updated the qat linear layer to maintain the
sparse_params which also solves the problem for the intrinsic qat layers.

Test Plan: python test/test_ao_sparsity.py TestQuantizedSparseKernels
python test/test_ao_sparsity.py TestQuantizedIntrinsicSparseKernels

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity] adding several sparse quantized layers"


Summary: This PR adds sparse+quantized versions of the following quantized layers

linear_relu
linear_relu (qat)
linear (qat)

also updated the default sparse+quantized mappings to utilize these new
layers and also updated the qat linear layer to maintain the
sparse_params which also solves the problem for the intrinsic qat layers.

Test Plan: python test/test_ao_sparsity.py TestQuantizedSparseKernels
python test/test_ao_sparsity.py TestQuantizedIntrinsicSparseKernels

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity] adding several sparse quantized layers"


Summary: This PR adds sparse+quantized versions of the following quantized layers

linear_relu
linear_relu (qat)
linear (qat)

also updated the default sparse+quantized mappings to utilize these new
layers and also updated the qat linear layer to maintain the
sparse_params which also solves the problem for the intrinsic qat layers.

Test Plan: python test/test_ao_sparsity.py TestQuantizedSparseKernels
python test/test_ao_sparsity.py TestQuantizedIntrinsicSparseKernels

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity] adding several sparse quantized layers"


Summary: This PR adds sparse+quantized versions of the following quantized layers

linear_relu
linear_relu (qat)
linear (qat)

also updated the default sparse+quantized mappings to utilize these new
layers and also updated the qat linear layer to maintain the
sparse_params which also solves the problem for the intrinsic qat layers.

Test Plan: python test/test_ao_sparsity.py TestQuantizedSparseKernels
python test/test_ao_sparsity.py TestQuantizedIntrinsicSparseKernels

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity] adding several sparse quantized layers"


Summary: This PR adds sparse+quantized versions of the following quantized layers

linear_relu
linear_relu (qat)
linear (qat)

also updated the default sparse+quantized mappings to utilize these new
layers and also updated the qat linear layer to maintain the
sparse_params which also solves the problem for the intrinsic qat layers.

Test Plan: python test/test_ao_sparsity.py TestQuantizedSparseKernels
python test/test_ao_sparsity.py TestQuantizedIntrinsicSparseKernels

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity] adding several sparse quantized layers"


Summary: This PR adds sparse+quantized versions of the following quantized layers

linear_relu
linear_relu (qat)
linear (qat)

also updated the default sparse+quantized mappings to utilize these new
layers and also updated the qat linear layer to maintain the
sparse_params which also solves the problem for the intrinsic qat layers.

Test Plan: python test/test_ao_sparsity.py TestQuantizedSparseKernels
python test/test_ao_sparsity.py TestQuantizedIntrinsicSparseKernels

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity] adding several sparse quantized layers"


Summary: This PR adds sparse+quantized versions of the following quantized layers

linear_relu
linear_relu (qat)
linear (qat)

also updated the default sparse+quantized mappings to utilize these new
layers and also updated the qat linear layer to maintain the
sparse_params which also solves the problem for the intrinsic qat layers.

Test Plan: python test/test_ao_sparsity.py TestQuantizedSparseKernels
python test/test_ao_sparsity.py TestQuantizedIntrinsicSparseKernels

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity] adding several sparse quantized layers"


Summary: This PR adds sparse+quantized versions of the following quantized layers

linear_relu
linear_relu (qat)
linear (qat)

also updated the default sparse+quantized mappings to utilize these new
layers and also updated the qat linear layer to maintain the
sparse_params which also solves the problem for the intrinsic qat layers.

Test Plan: python test/test_ao_sparsity.py TestQuantizedSparseKernels
python test/test_ao_sparsity.py TestQuantizedIntrinsicSparseKernels

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity] adding several sparse quantized layers"


Summary: This PR adds sparse+quantized versions of the following quantized layers

linear_relu
linear_relu (qat)
linear (qat)

also updated the default sparse+quantized mappings to utilize these new
layers and also updated the qat linear layer to maintain the
sparse_params which also solves the problem for the intrinsic qat layers.

Test Plan: python test/test_ao_sparsity.py TestQuantizedSparseKernels
python test/test_ao_sparsity.py TestQuantizedIntrinsicSparseKernels

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar commented on torch/ao/nn/sparse/intrinsic/qat/modules/linear_relu.py:
```diff
@@ -0,0 +1,51 @@
+    Note: There is no fused linear + relu for sparse quantized operation.
+    This module replaces it with sparse_quantized followed by torch.relu.
+
+    Ideally, we would want to inherit from the spars
+    Script doesn't like
```
nit: formatting seems off here

HDCharles replied:
fixed, good catch

supriyar reviewed (approved):
lgtm, @z-a-f could you take a look as well?

HDCharles committed:
Update on "[ao][sparsity] adding several sparse quantized layers"


Summary: This PR adds sparse+quantized versions of the following quantized layers

linear_relu
linear_relu (qat)
linear (qat)

also updated the default sparse+quantized mappings to utilize these new
layers and also updated the qat linear layer to maintain the
sparse_params which also solves the problem for the intrinsic qat layers.

Test Plan: python test/test_ao_sparsity.py TestQuantizedSparseKernels
python test/test_ao_sparsity.py TestQuantizedIntrinsicSparseKernels

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity] adding several sparse quantized layers"


Summary: This PR adds sparse+quantized versions of the following quantized layers

linear_relu
linear_relu (qat)
linear (qat)

also updated the default sparse+quantized mappings to utilize these new
layers and also updated the qat linear layer to maintain the
sparse_params which also solves the problem for the intrinsic qat layers.

Test Plan: python test/test_ao_sparsity.py TestQuantizedSparseKernels
python test/test_ao_sparsity.py TestQuantizedIntrinsicSparseKernels

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles commented:
@pytorchmergebot merge -g
